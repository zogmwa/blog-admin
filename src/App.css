html {
  box-sizing: border-box; /* 1 */
  cursor: default; /* 2 */
  -ms-text-size-adjust: 100%; /* 3 */
  -webkit-text-size-adjust: 100%; /* 3 */
}

.content {
  padding: 24px;
  margin: 24px;
  background: #fff;
}
