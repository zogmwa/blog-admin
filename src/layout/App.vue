<template>
  <div id="app">
    <transition name="fade">
      <router-view></router-view>
    </transition>
  </div>
</template>

<script>
import server from '../utils/axios'

export default {
  name: 'app',
  created () {
    server.get('/')
    .then(res => {
      console.log(res.data)
    })
  }
}
</script>

<style lang="scss">
@import '../assets/scss/variable.scss';

#app {
  position: absolute;
  top: 0px;
  bottom: 0px;
  width: 100%;
}
</style>
