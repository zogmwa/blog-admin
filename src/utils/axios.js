import axios from 'axios'
import * as config from '../config.js'
// import router from '../router'

const service = axios.create({
  baseURL: config.API_ROOT
})

// 拦截器
service.interceptors.request.use(config => {
  if (window.localStorage.getItem('TOKEN')) {
    config.headers.Authorization = window.localStorage.getItem('TOKEN')
  }
  return config
}, error => {
  return Promise.reject(error)
})

service.interceptors.response.use(response => {
  // if (response.data.code === 2) router.push('/login')
  return response
}, error => {
  return Promise.reject(error)
})

export default service
