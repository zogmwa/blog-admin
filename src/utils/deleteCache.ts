import { DataProxy } from 'apollo-cache';

// https://github.com/apollographql/apollo-feature-requests/issues/4#issuecomment-431119231
export const deleteCache = (cache: DataProxy, key: RegExp) => {
  // @ts-ignore
  Object.keys(cache.data.data).forEach(key => {
    // @ts-ignore
    key.match(key) && cache.data.delete(key);
  });
};
