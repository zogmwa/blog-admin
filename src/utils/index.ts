export { urlToList } from './pathToList';
export { checkLogin } from './isLogin';
export { deleteCache } from './deleteCache';
