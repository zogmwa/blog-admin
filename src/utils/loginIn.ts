export function loginIn () : boolean {
  if (!window.localStorage.getItem('TOKEN')) return false
  const lifeTime = JSON.parse(window.localStorage.getItem('TOKEN') || '').lifeTime * 1000
  const nowTime = (new Date()).getTime()
  if (nowTime > lifeTime) return false
  return true
}
