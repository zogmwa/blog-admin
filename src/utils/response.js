import { Notification } from 'element-ui'

export function error (message) {
  Notification.error({
    title: '错误',
    message
  })
}

export function success (message) {
  Notification.success({
    title: '成功',
    duration: 2000,
    message
  })
}
