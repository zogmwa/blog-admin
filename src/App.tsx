import React, { Suspense, ErrorInfo } from 'react';

import { ApolloProvider } from 'react-apollo';

import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Login from './pages/Login';
import BaseLayout from './layouts/BaseLayout';
import { PageLoading } from './components/PageLoading';
import { AuthRoute } from './router/auth';
import './App.css';
import './assets/scss/index.scss';
import { notification } from 'antd';
import { client } from './plugin/apollo';

export class App extends React.PureComponent {
  componentDidCatch(error: Error, info: ErrorInfo) {
    notification.error({
      message: 'something was error',
      description: info.componentStack,
      duration: 5
    });
  }

  render() {
    return (
      <BrowserRouter>
        <ApolloProvider client={client}>
          <Suspense fallback={<PageLoading />}>
            <Switch>
              <Route path="/login" exact component={Login} />
              <AuthRoute path="/" component={BaseLayout} />
            </Switch>
          </Suspense>
        </ApolloProvider>
      </BrowserRouter>
    );
  }
}
