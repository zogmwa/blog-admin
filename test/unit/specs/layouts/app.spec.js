import { shallow, createLocalVue } from 'vue-test-utils'
import Vuex from 'vuex'

import { destroyVm } from '../../utils/utils'
import App from '@/layout/App'

// 独立 Vue 构造函数
const localVue = createLocalVue()

localVue.use(Vuex)

describe('App.vue', () => {
  let actions
  let store

  let wrapper

  beforeEach(() => {
    actions = {
      initAuth: jest.fn() // 伪造函数
    }
    store = new Vuex.Store({
      state: {},
      actions
    })
  })

  afterEach(() => {
    destroyVm(wrapper.vm)
  })

  it('Call store action "initAuth" ', () => {
    wrapper = shallow(App, { store, localVue })
    expect(actions.initAuth).toHaveBeenCalled()
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
