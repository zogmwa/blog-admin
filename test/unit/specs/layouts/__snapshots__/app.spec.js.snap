// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue has the expected html structure 1`] = `
<div
  id="app"
>
  <router-view
    name="fade"
  />
</div>
`;
